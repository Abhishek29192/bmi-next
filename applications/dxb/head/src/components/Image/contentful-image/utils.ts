import type { ImageLoaderProps } from "next/image";
import type { Data, Options, Props } from "./types";

export const typeToObjectFitMap: {
  [key in NonNullable<Data["type"]>]: Options["size"];
} = {
  Decorative: "cover",
  Descriptive: "contain"
};

export const getPosition = ({
  size,
  position,
  focalPoint,
  isMobile
}: Options & Pick<Data, "focalPoint">): Options["position"] => {
  if (position) {
    return position;
  }

  if (!isMobile && size === "cover" && focalPoint) {
    return `${focalPoint.x}% ${focalPoint.y}%`;
  }

  return "center";
};

export const loader = ({ src, width }: ImageLoaderProps) => {
  return `${src}?w=${width}&q=50`;
};

export const getSizes = (widths: Props["widths"]) =>
  `(max-width: 599px) ${widths[0]}px, (max-width: 719px) ${widths[1]}px, (max-width: 839px) ${widths[2]}px, (max-width: 1439px) ${widths[3]}px, ${widths[4]}px`;
