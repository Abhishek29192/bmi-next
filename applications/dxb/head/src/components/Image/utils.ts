import { Props } from "./types";
import type { Props as ContentfulImageProps } from "./contentful-image/types";

export const isContentfulImage = (
  props: Props
): props is ContentfulImageProps =>
  "__typename" in props && props.__typename === "ContentfulImage";
