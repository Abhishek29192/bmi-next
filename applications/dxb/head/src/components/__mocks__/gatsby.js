"use strict";

const React = require("react");

const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(({ to, children, className }) =>
      React.createElement("a", { href: to, className }, children)
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
