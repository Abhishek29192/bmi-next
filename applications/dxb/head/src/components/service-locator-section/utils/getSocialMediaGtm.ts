import { GTM } from "../../../utils/google-tag-manager";
import { Service } from "../index";

export interface Props {
  service: Service;
  channel: string;
}

export const ID = "cta-share-ext-company";

export const getSocialMediaGtm = ({ channel, service }: Props): GTM => {
  const { address, certification, serviceTypes, name } = service;

  const roofProLevel = certification ? `- ${certification.toLowerCase()} ` : "";
  const serviceType = serviceTypes
    ? serviceTypes.map(({ name }) => name).join(", ")
    : "";

  const label = `${name} - ${address} ${roofProLevel}- ${serviceType} - ${channel}`;

  return {
    action: channel,
    id: ID,
    label
  };
};
