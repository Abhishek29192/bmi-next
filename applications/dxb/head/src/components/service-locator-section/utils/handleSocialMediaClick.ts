import { GTM, pushToDataLayer } from "../../../utils/google-tag-manager";
import { Service } from "../index";

export interface Props {
  service: Service;
  channel: string;
}

export const ID = "cta-share-ext-company";

export const handleSocialMediaClick = ({ channel, service }: Props) => {
  const { address, certification, entryType, name } = service;
  const roofProLevel = certification ? `- ${certification.toLowerCase()} ` : "";
  const label = `${name} - ${address} ${roofProLevel}- ${entryType} - ${channel}`;

  const gtm: GTM = {
    action: channel,
    id: ID,
    label
  };

  pushToDataLayer(gtm);

  // return true to allow anchor link to redirect
  return true;
};
