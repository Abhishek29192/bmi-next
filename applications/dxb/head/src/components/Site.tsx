import { createContext } from "react";
import { graphql } from "gatsby";
import { NavigationData } from "./Link";
import { Data as ResourcesData } from "./Resources";
import { GetMicroCopy, fallbackGetMicroCopy } from "./MicroCopy";

type Context = {
  node_locale: string;
  countryCode: string;
  homePage: {
    title: string;
  };
  getMicroCopy: GetMicroCopy;
  reCaptchaKey?: string;
  reCaptchaNet?: boolean;
};

export const SiteContext = createContext<Context>({
  node_locale: "",
  countryCode: "",
  homePage: {
    title: ""
  },
  getMicroCopy: fallbackGetMicroCopy
});

export type Data = {
  node_locale: string;
  homePage: {
    title: string;
  };
  countryCode: string;
  footerMainNavigation: NavigationData;
  footerSecondaryNavigation: NavigationData;
  menuNavigation: NavigationData;
  menuUtilities: NavigationData;
  resources: ResourcesData;
  headScripts?: {
    headScripts: string;
  } | null;
  scriptGA?: string | null;
  scriptOnetrust?: string | null;
  scriptHotJar?: string | null;
  scriptGOptLoad?: string | null;
};

export const query = graphql`
  fragment SiteFragment on ContentfulSite {
    node_locale
    homePage {
      title
    }
    countryCode
    menuNavigation {
      ...HeaderNavigationFragment
    }
    menuUtilities {
      ...HeaderUtilitiesFragment
    }
    footerMainNavigation {
      ...FooterMainNavigationFragment
    }
    footerSecondaryNavigation {
      ...FooterSecondaryNavigationFragment
    }
    resources {
      ...ResourcesFragment
    }
    headScripts {
      headScripts
    }
    scriptGA
    scriptOnetrust
    scriptHotJar
    scriptGOptLoad
  }
`;
