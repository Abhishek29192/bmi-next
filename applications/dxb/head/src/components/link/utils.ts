import { Link as GatsbyLink } from "gatsby";
import { getPathWithCountryCode } from "../../utils/path";
import { DataTypeEnum } from "./types";
import type { Data } from "./types";
import type { AnchorLinkActionProps } from "@bmi-digital/components/anchor-link";
import type { ButtonActionProps } from "@bmi-digital/components/button";
import type { Data as PageInfoData } from "../PageInfo";
import type { Data as PromoData } from "../Promo";

export const getCTA = (
  data: Pick<PromoData, "cta"> | Pick<PageInfoData, "path">,
  countryCode: string,
  linkLabel?: string
): AnchorLinkActionProps | null => {
  if ("cta" in data) {
    if (!data.cta) {
      return null;
    }

    return toAnchorLinkActionProps(data.cta, countryCode);
  }

  const { path } = data;
  if (!path) {
    return null;
  }

  const to = getPathWithCountryCode(countryCode, path).replace(/\/+/gi, "/");
  return {
    to: to,
    component: GatsbyLink,
    gtm: { id: "cta-click1", action: to, label: linkLabel }
  };
};

const internalHosts = process.env.GATSBY_INTERNAL_HOSTS?.split(",") || [
  "www.bmigroup.com"
];

export const isExternalUrl = (url: string): boolean => {
  try {
    const linkUrl = new URL(url);
    return !internalHosts.includes(linkUrl.host);
  } catch (e) {
    return false;
  }
};

const checkUrlAction = (url: string): boolean => {
  const actionUrls = ["mailto:", "tel:", "callto:"];
  return actionUrls.some((actionUrl) => url.startsWith(actionUrl));
};
export const toAnchorLinkActionProps = (
  link: Data,
  countryCode?: string
): AnchorLinkActionProps => {
  if (link.asset?.file?.url) {
    return {
      href: link.asset.file.url,
      download: true,
      external: true,
      gtm: {
        id: "cta-click1",
        action: link.asset.file.url,
        label: link.label
      }
    };
  }

  if (link.linkedPage?.path) {
    const to = getPathWithCountryCode(
      countryCode,
      link.linkedPage.path
    ).replace(/\/+/gi, "/");
    return {
      component: GatsbyLink,
      to,
      gtm: { id: "cta-click1", action: to, label: link.label }
    };
  }

  // TODO: DXB-7055 link.url _should_ be populated, but as any link can be used, it may not be
  return {
    href: link.url ?? undefined,
    external:
      (!!link.url && !checkUrlAction(link.url) && isExternalUrl(link.url)) ||
      undefined,
    gtm: { id: "cta-click1", action: link.url ?? undefined, label: link.label }
  };
};

export const toButtonActionProps = (
  link: Data,
  countryCode?: string,
  onClick?: ButtonActionProps["onClick"]
): ButtonActionProps => {
  if (link.type === DataTypeEnum.Visualiser) {
    return {
      onClick,
      gtm: {
        id: "cta-visualiser1",
        action: "visualiser",
        label: link.label
      }
    };
  }

  if (link.type === DataTypeEnum.Calculator) {
    return {
      onClick,
      gtm: {
        id: "cta-calculator1",
        action: "calculator",
        label: link.label
      }
    };
  }

  if (link.type === DataTypeEnum.Dialog) {
    return {
      onClick,
      gtm: { id: "cta-click1", action: link.type, label: link.label }
    };
  }

  return toAnchorLinkActionProps(link, countryCode);
};
