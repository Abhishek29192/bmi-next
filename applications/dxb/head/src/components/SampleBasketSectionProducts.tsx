import React from "react";
import { OverviewCard } from "@bmi/components";
import { Typography } from "@bmi/components";
import { Icon } from "@bmi/components";
import { Button } from "@bmi/components";
import { Remove } from "@material-ui/icons";
import { navigate } from "gatsby";
import { useMediaQuery, useTheme } from "@material-ui/core";
import { getPathWithCountryCode } from "../utils/path";
import { microCopy } from "../constants/microCopies";
import {
  ACTION_TYPES,
  Sample,
  useBasketContext
} from "../contexts/SampleBasketContext";
import styles from "./styles/SampleBasketSectionProducts.module.scss";
import { renderImage } from "./Image";
import { useSiteContext } from "./Site";
import { Classification } from "./types/pim";

interface Options {
  code: string;
  separator?: string;
  featureUnitRequired?: boolean;
}

export const getFeatures = (
  classifications: Classification[],
  options: Options
) => {
  const classification = classifications.find(
    (classification) =>
      classification.code.toLowerCase() === options.code.toLowerCase()
  );

  if (!classification) {
    return null;
  }

  const features = [
    ...new Set(
      classification?.features.flatMap((feature) =>
        feature.featureValues.map(({ value }) => value)
      )
    )
  ].filter((value) => value);

  const units =
    (options.featureUnitRequired &&
      classification?.features[0]?.featureUnit?.symbol) ||
    "";

  return `${features.join(options.separator || " | ")} ${units}`.trim();
};

const SampleBasketSectionProducts = () => {
  const { basketState, basketDispatch } = useBasketContext();
  const { getMicroCopy, countryCode } = useSiteContext();

  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("md"));

  const sampleCards = basketState.products.map((sample) => {
    const media = renderImage(
      {
        __typename: "ContentfulImage",
        type: "Descriptive",
        altText: sample.name,
        caption: { caption: sample.name },
        image: {
          file: {
            fileName: sample.name,
            url: sample.image
          }
        },
        focalPoint: { x: 0, y: 0 }
      },
      {
        className: styles["product-image"]
      }
    );

    const removeFromBasket = (payload: Sample) => {
      basketDispatch({
        type: ACTION_TYPES.BASKET_REMOVE,
        payload
      });
    };

    const handleRemove = (event: Event) => {
      event.stopPropagation();
      removeFromBasket(sample);
    };

    return (
      <OverviewCard
        className={styles["product-wrapper"]}
        key={sample.path}
        media={media}
        hasChildrenWithoutMargin
        onClick={() =>
          navigate(getPathWithCountryCode(countryCode, sample.path))
        }
        footer={
          !isMobile && (
            <Button
              className={styles["product-button"]}
              endIcon={<Remove />}
              onClick={handleRemove}
              variant="outlined"
            >
              {getMicroCopy(microCopy.PDP_OVERVIEW_REMOVE_SAMPLE)}
            </Button>
          )
        }
      >
        <div className={styles["product"]}>
          <div className={styles["product-description"]}>
            <Typography className={styles["product-title"]}>
              {sample.name}
            </Typography>
            <Typography className={styles["product-color"]}>
              {getFeatures(sample.classifications, {
                code: "appearanceAttributes"
              })}
            </Typography>
            <Typography className={styles["product-size"]}>
              {getFeatures(sample.classifications, {
                code: "measurements",
                separator: "x",
                featureUnitRequired: true
              })}
            </Typography>
          </div>
          {isMobile && (
            <Button
              className={styles["product-button"]}
              onClick={handleRemove}
              variant="outlined"
              isIconButton
            >
              <Icon source={Remove} />
            </Button>
          )}
        </div>
      </OverviewCard>
    );
  });

  return (
    <div className={styles["SampleBasketSectionProducts"]}>{sampleCards}</div>
  );
};

export default SampleBasketSectionProducts;
