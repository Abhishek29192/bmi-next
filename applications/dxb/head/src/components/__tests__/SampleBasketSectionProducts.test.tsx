import React from "react";
import * as Gatsby from "gatsby";
import { fireEvent, render, screen } from "@testing-library/react";
import { local } from "../../utils/storage";
import SampleBasketSectionProducts from "../SampleBasketSectionProducts";
import {
  BasketContextProvider,
  Sample
} from "../../contexts/SampleBasketContext";
import createImage from "../../__tests__/ImageHelper";
import createClassification, {
  createFeature
} from "../../__tests__/ClassificationHelper";
import { SiteContextProvider } from "../Site";

const getSiteContext = () => ({
  countryCode: "en",
  getMicroCopy: (microCopy: string) => `MC: ${microCopy}`,
  node_locale: "en-GB",
  homePage: {
    title: "Home page title"
  }
});

const samples: Sample[] = [
  {
    name: "sample-1",
    code: "sample-1",
    image: createImage().url,
    path: "sample-1-details",
    classifications: [createClassification({ code: "appearanceAttributes" })]
  },
  {
    name: "sample-2",
    code: "sample-2",
    path: "sample-2-details",
    image: createImage().url,
    classifications: [
      createClassification({
        code: "measurements",
        features: [
          createFeature({
            featureValues: [{ value: "10" }],
            featureUnit: { symbol: "mm", name: "mm", unitType: "size" }
          }),
          createFeature({ featureValues: [{ value: "20" }] })
        ]
      })
    ]
  },
  {
    name: "sample-3",
    code: "sample-3",
    path: "sample-3-details",
    image: createImage().url,
    classifications: []
  }
];

describe("SampleBasketSectionProducts component", () => {
  beforeAll(() => {
    jest.spyOn(local, "getItem").mockReturnValue(JSON.stringify(samples));
  });

  it("renders correctly", () => {
    const { container } = render(
      <BasketContextProvider>
        <SampleBasketSectionProducts />
      </BasketContextProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should remove sample", () => {
    render(
      <BasketContextProvider>
        <SampleBasketSectionProducts />
      </BasketContextProvider>
    );

    fireEvent.click(screen.queryAllByText("MC: pdp.overview.removeSample")[0]);

    expect(screen.queryByText("sample-1")).toBeNull();
  });

  it("should navigate to details", () => {
    jest.spyOn(Gatsby, "navigate").mockImplementation();

    render(
      <SiteContextProvider value={getSiteContext()}>
        <BasketContextProvider>
          <SampleBasketSectionProducts />
        </BasketContextProvider>
      </SiteContextProvider>
    );

    fireEvent.click(screen.queryByText("sample-1"));

    expect(Gatsby.navigate).toBeCalledWith("/en/sample-1-details");
  });

  it("renders correctly for mobile", () => {
    jest.mock("@material-ui/core", () => ({
      useMediaQuery: jest.fn().mockRejectedValue(true)
    }));

    const { container } = render(
      <BasketContextProvider>
        <SampleBasketSectionProducts />
      </BasketContextProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
