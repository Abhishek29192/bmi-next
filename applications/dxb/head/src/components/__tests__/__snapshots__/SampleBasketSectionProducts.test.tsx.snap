// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleBasketSectionProducts component renders correctly 1`] = `
<div>
  <div
    class="SampleBasketSectionProducts"
  >
    <button
      class="MuiButtonBase-root OverviewCard-clickable product-wrapper MuiBox-root css-lx6wpa-MuiButtonBase-root"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-1"
          class="product-image OverviewCard-headerPicture css-7mgsqw-Media"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="product"
            >
              <div
                class="product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-1
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="OverviewCard-clickable css-1cyx9t4"
        data-testid="overview-card-body-footer"
      >
        <button
          class="MuiButtonBase-root MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium product-button css-1ga0prt-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          MC: pdp.overview.removeSample
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root OverviewCard-clickable product-wrapper MuiBox-root css-lx6wpa-MuiButtonBase-root"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-2"
          class="product-image OverviewCard-headerPicture css-7mgsqw-Media"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="product"
            >
              <div
                class="product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-2
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  Red | Matte
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  10x20x30 mm
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="OverviewCard-clickable css-1cyx9t4"
        data-testid="overview-card-body-footer"
      >
        <button
          class="MuiButtonBase-root MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium product-button css-1ga0prt-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          MC: pdp.overview.removeSample
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root OverviewCard-clickable product-wrapper MuiBox-root css-lx6wpa-MuiButtonBase-root"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-3"
          class="product-image OverviewCard-headerPicture css-7mgsqw-Media"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="product"
            >
              <div
                class="product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-3
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  Black
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="OverviewCard-clickable css-1cyx9t4"
        data-testid="overview-card-body-footer"
      >
        <button
          class="MuiButtonBase-root MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root Button-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium product-button css-1ga0prt-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          MC: pdp.overview.removeSample
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
